### PR TITLE
feat(l1): new pooled transaction hashes

### LINKED ISSUE #383: NewPooledTransactionHashes (0x08)

### PR DESCRIPTION
**Pull request by Arkenan**
Changes:
- Parses and handles new pooled transactions incoming message.
- Sends a get pooled transactions for missing transactions. Checks the mempool to not ask for already present ones.

This also adds the NewPooledTx hive test. The next PR will add support for the PooledTransactions response so we can fully exchange mempools with other nodes + receive blob transactions via p2p.

Closes #383

### CODE REVIEW

File: crates/storage/store/storage.rs [hunk 1]
```diff
@@ -325,6 +325,24 @@ impl Store {
         Ok(txs_by_sender)
     }
 
+    /// Gets hashes from possible_hashes that are not already known in the mempool.
+    pub fn get_unknown_transactions(
```
Comment (fkrause98) [inline crates/storage/store/storage.rs:329]: Nit:
get_unknown_transactions sounds a bit weird to me.
I think I would maybe call this 'filter_known_transactions', what do you think?

**Review by fkrause98 — APPROVED** [review]

File: crates/networking/p2p/rlpx/connection.rs [hunk 2]
```diff
@@ -377,6 +379,17 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
                 };
                 self.send(Message::BlockBodies(response)).await?;
             }
+            Message::NewPooledTransactionHashes(new_pooled_transaction_hashes)
+                if peer_supports_eth =>
+            {
+                // For now we always ask for the transactions we don't know. We might want to add
```
Comment (fkrause98) [inline crates/networking/p2p/rlpx/connection.rs:385]: Also, maybe add an issue for this?

File: crates/storage/store/storage.rs (hunk 1 above)
Comment (Arkenan) [reply crates/storage/store/storage.rs:329]: Done

File: crates/networking/p2p/rlpx/connection.rs (hunk 2 above)
Comment (Arkenan) [reply crates/networking/p2p/rlpx/connection.rs]: Done

**Pull request merged**